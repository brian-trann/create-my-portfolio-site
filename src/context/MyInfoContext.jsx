import React from 'react';
const MyInfoContext = React.createContext();
export default MyInfoContext;
