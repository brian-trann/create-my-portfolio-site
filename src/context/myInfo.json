{
    "navigation": {
        "name": "John Doe",
        "menu": [
            {
                "text": "Home",
                "url": "/",
                "target":""
            },
            {
                "text": "About",
                "url": "/about",
                "target":""
            },
            {
                "text": "Github",
                "url": "https://github.com/",
                "target":"_blank"
            },
            {
                "text": "LinkedIn",
                "url": "https://www.linkedin.com//",
                "target":"_blank"
            }
        ]
    },
    "main": {
        "greeting": "Hi, I'm John, a Software Developer from California",
        "blurb":"Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sapien tortor, ornare nec leo nec, convallis pulvinar tortor. Ut lorem sem, tempus eu cursus eu, pulvinar sit amet neque.",
        "profileImage": { 
            "url":"",
            "altText":"profile image"
        },
        "image": {
            "url": "/",
            "altText": "temp"
        }
    },
    "work":{
        "categoryName":"Selected Work",
        "projects":[
            {
                "projectName": "project0",
                "description":"Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sapien tortor, ornare nec leo nec, convallis pulvinar tortor. Ut lorem sem, tempus eu cursus eu, pulvinar sit amet neque.",
                "githubUrl":"https://github.com/",
                "projectUrl":"https://vercel.com/",
                "image": {
                    "url": "https://github.githubassets.com/images/modules/site/home/repo-browser.png",
                    "altText": "project0 thumbnail"
                }
            },
            {
                "projectName": "project1",
                "description":"Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sapien tortor, ornare nec leo nec, convallis pulvinar tortor. Ut lorem sem, tempus eu cursus eu, pulvinar sit amet neque.",
                "githubUrl":"https://github.com/",
                "projectUrl":"https://vercel.com/",
                "image": {
                    "url": "https://github.githubassets.com/images/modules/site/home/repo-browser.png",
                    "altText": "project1 thumbnail"
                }
            },
            {
                "projectName": "project2",
                "description":"Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sapien tortor, ornare nec leo nec, convallis pulvinar tortor. Ut lorem sem, tempus eu cursus eu, pulvinar sit amet neque.",
                "githubUrl":"https://github.com/",
                "projectUrl":"https://vercel.com/",
                "image": {
                    "url": "https://github.githubassets.com/images/modules/site/home/repo-browser.png",
                    "altText": "project2 thumbnail"
                }
            }
        ],
        "buttonNames":{
            "primary":"Github",
            "secondary":"Site"
        }
    },
    "about": {
        "categoryName":"About",
        "statement": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sapien tortor, ornare nec leo nec, convallis pulvinar tortor. Ut lorem sem, tempus eu cursus eu, pulvinar sit amet neque.",
        "skills":{
            "categoryName":"Skills",
            "items":[
                {
                    "type":"Languages",
                    "items":["JavaScript", "HTML", "CSS","Python","SQL"]
                },
                {
                    "type":"Libraries and Frameworks",
                    "items":["React", "Redux", "Express","Flask"]
                },
                {
                    "type":"Other",
                    "items":["other0", "other1"]
                }
            ]
        },
        "workExperience": {
            "categoryName":"Work Experience",
            "work": [
                {
                    "company":"company0",
                    "position":"position0",
                    "date":"Date 2020 - Date 2021",
                    "description":"Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sapien tortor, ornare nec leo nec, convallis pulvinar tortor. Ut lorem sem, tempus eu cursus eu, pulvinar sit amet neque."
                },
                {
                    "company":"company2",
                    "position":"position1",
                    "date":"Date 2019 - Date 2020",
                    "description":"Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sapien tortor, ornare nec leo nec, convallis pulvinar tortor. Ut lorem sem, tempus eu cursus eu, pulvinar sit amet neque."
                }
            ]
        },
        "education":{
            "categoryName":"Education",
            "items": [
                {
                    "name":"education0",
                    "curriculum":"curriculum0",
                    "date":"Date 2020 - Date 2021",
                    "description":"Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sapien tortor, ornare nec leo nec, convallis pulvinar tortor. Ut lorem sem, tempus eu cursus eu, pulvinar sit amet neque."
                }
            ]
        },
        "certifications":{
            "categoryName":"Certifications",
            "items":["certification0", "certification1"]
        }
    },
    "other":{
        "images":[
            {"image":{
                "url":"https://github.githubassets.com/images/modules/site/home/repo-browser.png",
                "altText":"",
                "description":""
                }
            }
        ]
    }
}