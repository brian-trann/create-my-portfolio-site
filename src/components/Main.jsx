import React, { useContext } from 'react';
import MyInfoContext from '../context/MyInfoContext';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import Container from '@material-ui/core/Container';
import Divider from '@material-ui/core/Divider';
import sampleImg from '../sampleImg.jpg';
import Grid from '@material-ui/core/Grid';
import CardActions from '@material-ui/core/CardActions';
import Button from '@material-ui/core/Button';

const useStyles = makeStyles((theme) => ({
	main           : {
		padding      : theme.spacing(8, 0, 6),
		marginBottom : theme.spacing(8)
	},
	cardsTitle     : {
		paddingBottom : theme.spacing(6)
	},
	cardGrid       : {
		paddingTop    : theme.spacing(8),
		paddingBottom : theme.spacing(8)
	},
	card           : {
		height        : '100%',
		display       : 'flex',
		flexDirection : 'column'
	},
	cardMedia      : {
		paddingTop : '56.25%'
	},
	cardContent    : {
		flexGrow : 1
	},
	profilePicture : {
		float       : 'right',
		marginRight : '2rem'
	},
	mainBlurb      : {
		paddingBottom : theme.spacing(2)
	}
}));
const Main = () => {
	const classes = useStyles();
	const { main, work } = useContext(MyInfoContext);
	return (
		<React.Fragment>
			{/** Main unit */}
			<div className={classes.main}>
				<Container maxWidth='md'>
					<div className={classes.profilePicture}>
						<img
							width='300px'
							src={main.profileImage.url || sampleImg}
							alt={main.profileImage.altText}
						/>
					</div>
					<Typography
						component='h1'
						variant='h2'
						align='left'
						color='textPrimary'
						gutterBottom
					>
						{main.greeting}
					</Typography>
				</Container>
			</div>

			{/** End Main unit */}
			{/** Secondary unit */}
			<Container maxWidth='md'>
				<Typography
					className={classes.mainBlurb}
					variant='subtitle1'
					align='center'
					color='textSecondary'
					gutterBottom
				>
					{main.blurb}
				</Typography>

				<Divider />
			</Container>

			{/** End Secondary unit */}
			{/** Cards unit */}

			<Container maxWidth='md' className={classes.cardGrid}>
				<div className={classes.cardsTitle}>
					<Typography
						variant='h5'
						component='h3'
						align='center'
						color='textSecondary'
						gutterBottom
					>
						{work.categoryName}
					</Typography>
				</div>
				<Grid container spacing={4}>
					{work.projects.map((w) => (
						<Grid item key={w.projectName} xs={12} sm={6} md={4}>
							<Card className={classes.card}>
								<CardMedia
									className={classes.cardMedia}
									image={w.image.url}
									title={w.image.altText}
								/>
								<CardContent className={classes.cardContent}>
									<Typography gutterBottom variant='h5' component='h2'>
										{w.projectName}
									</Typography>

									<Typography variant='body2'>{w.description}</Typography>
								</CardContent>
								<CardActions>
									<Button
										size='small'
										color='primary'
										onClick={() => window.open(w.url)}
									>
										{work.buttonName}
									</Button>
								</CardActions>
							</Card>
						</Grid>
					))}
				</Grid>
			</Container>
			{/** End Cards unit */}
		</React.Fragment>
	);
};
export default Main;
