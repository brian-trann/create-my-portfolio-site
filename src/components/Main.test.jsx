import React from 'react';
import { render } from '@testing-library/react';
import Main from './Main';
import myInfo from '../context/myInfo.json';
import MyInfoContext from '../context/MyInfoContext';

// smoke test
it('renders without crashing', () => {
	render(
		<MyInfoContext.Provider value={myInfo}>
			<Main />
		</MyInfoContext.Provider>
	);
});

// snapshot test
it('matches snapshot', () => {
	const { asFragment } = render(
		<MyInfoContext.Provider value={myInfo}>
			<Main />
		</MyInfoContext.Provider>
	);
	expect(asFragment()).toMatchSnapshot();
});
