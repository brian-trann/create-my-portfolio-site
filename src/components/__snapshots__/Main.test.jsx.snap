// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-main-9"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthMd"
    >
      <div
        class="makeStyles-profilePicture-15"
      >
        <img
          alt="profile image"
          src="genericUser.svg"
          width="300px"
        />
      </div>
      <h1
        class="MuiTypography-root MuiTypography-h2 MuiTypography-colorTextPrimary MuiTypography-gutterBottom MuiTypography-alignLeft"
      >
        Hi, I'm John, a Software Developer from California
      </h1>
    </div>
  </div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthMd"
  >
    <h6
      class="MuiTypography-root makeStyles-mainBlurb-16 MuiTypography-subtitle1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom MuiTypography-alignCenter"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sapien tortor, ornare nec leo nec, convallis pulvinar tortor. Ut lorem sem, tempus eu cursus eu, pulvinar sit amet neque.
    </h6>
    <hr
      class="MuiDivider-root"
    />
  </div>
  <div
    class="MuiContainer-root makeStyles-cardGrid-11 MuiContainer-maxWidthMd"
  >
    <div
      class="makeStyles-cardsTitle-10"
    >
      <h3
        class="MuiTypography-root MuiTypography-h5 MuiTypography-colorTextSecondary MuiTypography-gutterBottom MuiTypography-alignCenter"
      >
        Selected Work
      </h3>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
      >
        <div
          class="MuiPaper-root MuiCard-root makeStyles-card-12 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            class="MuiCardMedia-root makeStyles-cardMedia-13"
            style="background-image: url(https://github.githubassets.com/images/modules/site/home/repo-browser.png);"
            title="project0 thumbnail"
          />
          <div
            class="MuiCardContent-root makeStyles-cardContent-14"
          >
            <h2
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
            >
              project0
            </h2>
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sapien tortor, ornare nec leo nec, convallis pulvinar tortor. Ut lorem sem, tempus eu cursus eu, pulvinar sit amet neque.
            </p>
          </div>
          <div
            class="MuiCardActions-root MuiCardActions-spacing"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Go
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
      >
        <div
          class="MuiPaper-root MuiCard-root makeStyles-card-12 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            class="MuiCardMedia-root makeStyles-cardMedia-13"
            style="background-image: url(https://github.githubassets.com/images/modules/site/home/repo-browser.png);"
            title="project1 thumbnail"
          />
          <div
            class="MuiCardContent-root makeStyles-cardContent-14"
          >
            <h2
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
            >
              project1
            </h2>
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sapien tortor, ornare nec leo nec, convallis pulvinar tortor. Ut lorem sem, tempus eu cursus eu, pulvinar sit amet neque.
            </p>
          </div>
          <div
            class="MuiCardActions-root MuiCardActions-spacing"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Go
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
      >
        <div
          class="MuiPaper-root MuiCard-root makeStyles-card-12 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            class="MuiCardMedia-root makeStyles-cardMedia-13"
            style="background-image: url(https://github.githubassets.com/images/modules/site/home/repo-browser.png);"
            title="project2 thumbnail"
          />
          <div
            class="MuiCardContent-root makeStyles-cardContent-14"
          >
            <h2
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
            >
              project2
            </h2>
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sapien tortor, ornare nec leo nec, convallis pulvinar tortor. Ut lorem sem, tempus eu cursus eu, pulvinar sit amet neque.
            </p>
          </div>
          <div
            class="MuiCardActions-root MuiCardActions-spacing"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Go
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
